### DISCUSSION TITLE
SurrealDB or Cozodb

### DISCUSSION THREAD

**Discussion by demosthenez** [sqlpage/SQLPage#198] [Ideas]
Just an idea: I think it would be cool to see one of these rust based graph db projects paired with something like SQLpage.  Just an idea.  I'm aware this project is using some kind of ORM or something with it's own limitations, so this idea might be a bit far removed from reality.  On a side note, there seem to be limitations with using SQLite FTS and this existing ORM.  It doesn't seem to recognize/pass certain commands such as MATCH for basic FTS usage.

**Comment by lovasoa** [comment]
Hi and welcome to SQLPage!

If you want to add support for new databases, you can contribute support for them in the https://github.com/lovasoa/sqlx-oldapi library.  For more niche databases like the ones you mention, it may be a more generally useful to contribute support for an existing binary protocol such as the postgres wire protocol to them. Anyway, this is quite a bit of work, and not something I currently prioritize myself. 

For existing SQL syntax in already supported databases that does not work in SQLPage, I consider it a bug and am willing to fix it myself. Feel free to open issues here or directly open a pull request in https://github.com/sqlparser-rs/sqlparser-rs and mention me.
